src/parser/mod.rs

```diff
@@ -5723,16 +5734,19 @@ impl<'a> Parser<'a> {
 
         let strict = self.parse_keyword(Keyword::STRICT);
 
-        let comment = if self.parse_keyword(Keyword::COMMENT) {
-            let _ = self.consume_token(&Token::Eq);
-            let next_token = self.next_token();
-            match next_token.token {
-                Token::SingleQuotedString(str) => Some(CommentDef::WithoutEq(str)),
-                _ => self.expected("comment", next_token)?,
-            }
-        } else {
-            None
-        };
+        // Excludes Hive dialect here since it has been handled after table column definitions.
+        if !dialect_of!(self is HiveDialect) {
+            comment = if self.parse_keyword(Keyword::COMMENT) {
```

Comment:
we can probably merge this with the dialect check to be `if !dialect_of() && self.parse_keyword()`? I think that lets us skip the `else { None }` block

Reply:
Good point, has been changed according to the suggestion.